crates/core_simd/src/vector.rs

```diff
@@ -979,3 +1189,37 @@ where
 {
     type Mask = isize;
 }
+
+#[inline]
+fn lane_indices<const N: usize>() -> Simd<usize, N>
+where
+    LaneCount<N>: SupportedLaneCount,
+{
+    let mut index = [0; N];
+    for i in 0..N {
+        index[i] = i;
+    }
+    Simd::from_array(index)
+}
+
+#[inline]
+fn mask_up_to<M, const N: usize>(len: usize) -> Mask<M, N>
+where
+    LaneCount<N>: SupportedLaneCount,
+    M: MaskElement,
+{
+    let index = lane_indices::<N>();
+    let max_value: u64 = M::max_unsigned();
+    macro_rules! case {
+        ($ty:ty) => {
+            if N < <$ty>::MAX as usize && max_value as $ty as u64 == max_value {
```

Comment:
```suggestion
            if N < <$ty>::MAX as usize && max_value == <$ty>::MAX as u64 {
```

Is there a difference?

Reply:
yes those are not equivalent, `max_value as $ty as u64 == max_value` is `true` when `M = i8` and `$ty = u16` which is what you'd get with `Mask<i8, 256>`. basically it's checking `$ty::BITS >= M::BITS`

Reply:
I don't think I can write a test for it until SupportedLaneCount is increased

Reply:
> I don't think I can write a test for it until SupportedLaneCount is increased

yeah, that's fine. if it's broken we can fix it later, imo the important part is that we tried to not bake in things we know will be broken.